pyanaconda/modules/payloads/payload/dnf/dnf.py

```diff
@@ -43,16 +47,40 @@ def type(self):
     @property
     def supported_source_types(self):
         """Get list of sources supported by DNF module."""
-        # TODO: Add supported sources when implemented
-        return None
+        return [
+            SourceType.CDROM,
+            SourceType.HDD,
+            SourceType.HMC,
+            SourceType.NFS,
+            SourceType.REPO_FILES,
+            SourceType.URL
+        ]
 
     def process_kickstart(self, data):
         """Process the kickstart data."""
-        pass
+        source_types = SourceFactory.get_type_for_kickstart(data)
+
+        if len(source_types) > 1:
+            types_strs = map(lambda x: "\"{}\"".format(x.ks_data.currentCmd), source_types)
+            msg = "Only one of {} payload sources can be used!".format(", ".join(types_strs))
+            lineno = source_types[-1].ks_data.lineno
+            raise KickstartParseError(msg, lineno=lineno)
```

Comment:
This check should be somehow included in the function `get_type_for_kickstart` or moved to a new function or to `parse` methods of the kickstart commands in `pyanaconda/modules/payloads/kickstart.py` or to pykickstart. This solution is scary.

Reply:
The same as above. I've tried a different approach.